src/snowflake/snowpark/_internal/udf_utils.py

```diff
@@ -382,6 +385,79 @@ def get_types_from_type_hints(
     return return_type, input_types
 
 
+def get_opt_arg_defaults(
+    func: Union[Callable, Tuple[str, str]],
+    object_type: TempObjectType,
+    input_types: List[DataType],
+) -> List[Optional[str]]:
+    EMPTY_DEFAULT_VALUES = [None] * len(input_types)
+
+    def build_default_values_result(
+        default_values: Any,
+        input_types: List[DataType],
+        convert_python_str_to_object: bool,
+    ) -> List[Optional[str]]:
+        num_optional_args = len(default_values)
+        num_positional_args = len(input_types) - num_optional_args
```

Comment:
I remember python supports the following function definition
```
def foo(a, b=1, *, c=2)
```

in this case both `b` is a positional argument with default value, while `c` is a keyword argument with default value.
is it a legit case in sproc/udf? and if it's legit, does `default_values` cover the positional argument default value?

Reply:
this is not supported currently and we cannot enforce this behavior for sql. I say we skip this support for now unless it is asked by customers

Reply:
sounds good, thanks for checking